qiskit_nature/second_q/operators/bosonic_op.py

```diff
@@ -489,7 +485,7 @@ def is_hermitian(self, atol: float | None = None) -> bool:
         if self.is_parameterized():
             raise ValueError("is_hermitian is not supported for operators containing parameters.")
         atol = self.atol if atol is None else atol
-        diff = (self - self.adjoint()).simplify(atol=atol)
+        diff = (self - self.adjoint()).index_order().simplify(atol=atol)
```

Comment:
```suggestion
        diff = (self - self.adjoint()).normal_order().simplify(atol=atol)
```
At least that is what we do in `FermionicOp`. Does this cause any problems?